### ISSUE TITLE
Manual Controls Issue

### ISSUE THREAD

**Issue by samgoldman** [OHSBlazerbots/Stronghold2016#2]
So of course I thought of this just now, but you know how when manually controlling the arm, it would do pretty well and then slam down and stall? Well if you think like the computer, it is doing this:

```
1. Check if joystick value isn't 0
2. Add the appropriate value to the setpoint
3. Repeat 1-3 every 20ms
```

Because it is doing this so quickly (every 20ms) a short tap of the joystick could add upwards of 100 degrees to the setpoint, oviously now what we want. You could decrease the number of degrees added each time but that is only treating the symptom.
A proper fix would be to do this:

```
1. Check if joystick value isn't 0
2. Check if the arm is on target*
3.1. If the arm is on target, ad the appropriate value to the setpoint
3.2. Else, forget it, lets get to the target before we start thinking ahead to moving further
4. Repeat 1-4 every 20ms 
5. Enjoy**
```

*[Use this](http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/command/PIDSubsystem.html#onTarget--)
**If you notice the infinite loop, you'll realize that 'enjoy' occurs when the time stops and 4 is no longer relevant and the loop is finished

**Comment by melodypanda** [comment]
OH! That makes a lot of sense! I will implement that ASAP.
